docs/deployment-aws.md

```diff
@@ -0,0 +1,202 @@
+```
+### 2. Install [Terraform][2] CLI
+```bash
+sudo apt-get update && sudo apt-get install -y gnupg software-properties-common curl
+curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
+sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
+sudo apt-get update && sudo apt-get install terraform
+```
+
+### 3. Install `Python` and `pip`
+```bash
+sudo apt-get update
+sudo apt-get install -y --no-install-recommends python3
+sudo apt install python3-pip
+sudo apt install python3-testresources
+```
+
+### 4. Install [Ansible][4] and its dependencies
+    
+Run the following commands from the [DCL][5] project home
+```bash
+sudo pip install -r deployment/requirements.txt
+ansible-galaxy install -r deployment/galaxy-requirements.yml 
+```
+
+### 5. Install [Dasel][6] to make conversions between `json` and `yaml`
+```bash
+sudo wget -qO /usr/local/bin/dasel https://github.com/TomWright/dasel/releases/latest/download/dasel_linux_amd64
+sudo chmod a+x /usr/local/bin/dasel
+```
+
+## Terraform and Ansible Configuration
+### 1. Set up an AWS user for use with Terraform
+    
+Create credentials file [`~/.aws/credentials`] with the following content:
+```text
+[default]
+aws_access_key_id = <access_key_id_here>
+aws_secret_access_key = <secret_access_key_here>
+```
+> **_Note:_** Your account must have enough privileges to manage all AWS resources required by terraform
+
+### 2. Disable host key checking for Ansible (to avoid host key checking when Ansible connects to AWS instances using ssh)
+
+Create Ansible configuration file [`~/.ansible.cfg`] with the following content:
+```
+ANSIBLE_HOST_KEY_CHECKING=False
+```
+
+## Deployment Configuration
+
+### 1. Configure AWS infrastructure parameters in [`deployment/terraform/aws/terraform.tfvars`]
+    
+### AWS Regions:
+```
+region_1 = "us-west-1"
+region_2 = "us-east-2"
+```
+- Selects two regions where nodes will be created
+
+### Validator:
+```
+validator_config = {
+    instance_type = "t3.medium"
+}
+```
+- Validator node is created in `region_1` by default
+
+
+### Private Sentries:
+
+```hcl
+private_sentries_config = {
+    enable        = true
+    nodes_count   = 2
+    instance_type = "t3.medium"
+}
+```
+- Private sentry nodes are created in the region as Validator by default
+- Can be disabled by setting `enable = false`
+- Only one instance of private sentry is created with static ip address
+
+### Public Sentries:
+```hcl
+public_sentries_config = {
+    enable        = true
+    enable_ipv6   = false
+    nodes_count   = 2
+    instance_type = "t3.medium"
+
+    regions = [
+        1,
+        2
+    ]
+}
+```
+- Requires `Private Sentries` being enabled
+- Can be configured to have IPv6 addresses using `enable_ipv6 = true`
+    > **_Note:_** Number of available IPv4 static addresses is restricted to 5 per region on AWS by default
+- Can be configured to run on multiple regions
+
+### Observers:
```

Comment:
I think we need to clearly describe how to setup different use cases, and how to enable/disable optional components:


1. Genesis Validator + optional Sentries + optional Observers
2. Validator + optional Sentries + optional Observers
3. Observers only